bug: flake: quota

### Describe the bug

```
 === FAIL: test/e2e/quota TestKubeQuotaNormalCRDs (unknown)
    quota_test.go:252: shared kcp server will target configuration "/go/src/github.com/kcp-dev/kcp/.kcp/admin.kubeconfig"
    quota_test.go:268: Created root:organization workspace root:e2e-workspace-cx2ww as /clusters/1zcegg5d1igixo6t
    quota_test.go:275: Created root:universal workspace root:e2e-workspace-cx2ww:one as /clusters/30ln42srrbq4011n
    quota_test.go:276: Created root:universal workspace root:e2e-workspace-cx2ww:two as /clusters/224pc4kizldywhnr
    quota_test.go:278: Install a normal sheriffs CRD into workspace 1 "root:e2e-workspace-cx2ww:one"
I0126 19:16:42.920442   30739 bootstrap.go:129] "bootstrapping CRD" crd="sheriffs.aj8y3ikb.io"
    quota_test.go:281: Install another normal sheriffs CRD with a different schema into workspace 2 "root:e2e-workspace-cx2ww:two"
I0126 19:16:43.124610   30739 bootstrap.go:129] "bootstrapping CRD" crd="sheriffs.aj8y3ikb.io"
I0126 19:16:43.145999   30739 bootstrap.go:150] "bootstrapped CRD" crd="sheriffs.aj8y3ikb.io" customresourcedefinition.workspace="224pc4kizldywhnr" customresourcedefinition.namespace="" customresourcedefinition.name="sheriffs.aj8y3ikb.io" customresourcedefinition.apiVersion="" duration="21.344088ms"
I0126 19:16:43.146095   30739 bootstrap.go:169] "waiting for CRD to be established" crd="sheriffs.aj8y3ikb.io" customresourcedefinition.workspace="224pc4kizldywhnr" customresourcedefinition.namespace="" customresourcedefinition.name="sheriffs.aj8y3ikb.io" customresourcedefinition.apiVersion=""
    quota_test.go:303: Creating ws 1 quota
    quota_test.go:307: Waiting for ws 1 quota to show usage
    quota_test.go:316: Create 2 sheriffs to reach the quota limit
    quota_test.go:317: Creating aj8y3ikb.io/v1 sheriffs root:e2e-workspace-cx2ww:one|default/ws1-1
    quota_test.go:318: Creating aj8y3ikb.io/v1 sheriffs root:e2e-workspace-cx2ww:one|default/ws1-2
    quota_test.go:320: Make sure quota is enforcing limits
    quota_test.go:324: Trying to create a sheriff
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x5a69348]
goroutine 2936 [running]:
github.com/kcp-dev/kcp/test/e2e/quota.TestKubeQuotaNormalCRDs.func2()
	/go/src/github.com/kcp-dev/kcp/test/e2e/quota/quota_test.go:328 +0x588
github.com/kcp-dev/kcp/test/e2e/framework.Eventually.func1()
	/go/src/github.com/kcp-dev/kcp/test/e2e/framework/util.go:330 +0x88
github.com/stretchr/testify/assert.Eventually.func1()
	/go/pkg/mod/github.com/stretchr/testify@v1.7.1/assert/assertions.go:1691 +0x3a
created by github.com/stretchr/testify/assert.Eventually
	/go/pkg/mod/github.com/stretchr/testify@v1.7.1/assert/assertions.go:1691 +0x3a6 
```

### Steps To Reproduce

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/kcp-dev_kcp/2683/pull-ci-kcp-dev-kcp-main-e2e-sharded/1618687639924248576

### Expected Behaviour

.

### Additional Context

@ncdc

--- a/test/e2e/quota/quota_test.go
+++ b/test/e2e/quota/quota_test.go
@@ -100,7 +100,7 @@ func TestKubeQuotaBuiltInCoreV1Types(t *testing.T) {
 			if _, err := kubeClusterClient.Cluster(wsPath).CoreV1().ConfigMaps("default").Create(ctx, cm, metav1.CreateOptions{}); err != nil {
 				return apierrors.IsForbidden(err), err.Error()
 			}
-			return false, ""
+			return false, "expected an error trying to create a configmap"
 		}, wait.ForeverTestTimeout, 100*time.Millisecond, "quota never rejected configmap creation")
 	}
 }
@@ -239,7 +239,7 @@ func TestKubeQuotaCoreV1TypesFromBinding(t *testing.T) {
 				if err != nil {
 					return apierrors.IsForbidden(err), err.Error()
 				}
-				return true, ""
+				return false, "expected an error trying to create a service"
 			}, wait.ForeverTestTimeout, 100*time.Millisecond, "quota never rejected service creation")
 		})
 	}
@@ -325,7 +325,10 @@ func TestKubeQuotaNormalCRDs(t *testing.T) {
 			sheriff := NewSheriff(group, fmt.Sprintf("ws%d-%d", wsIndex, i))
 			i++
 			_, err := dynamicClusterClient.Cluster(ws).Resource(sheriffsGVR).Namespace("default").Create(ctx, sheriff, metav1.CreateOptions{})
-			return apierrors.IsForbidden(err), err.Error()
+			if err != nil {
+				return apierrors.IsForbidden(err), err.Error()
+			}
+			return false, "expected an error trying to create a sheriff"
 		}, wait.ForeverTestTimeout, 100*time.Millisecond, "quota never rejected sheriff creation")
 	}
 }
@@ -426,7 +429,7 @@ func TestClusterScopedQuota(t *testing.T) {
 			if err != nil {
 				return apierrors.IsForbidden(err), err.Error()
 			}
-			return true, ""
+			return false, "expected an error trying to create a configmap"
 		}, wait.ForeverTestTimeout, 100*time.Millisecond, "quota never rejected configmap creation in %q", wsPath)
 
 		t.Logf("Make sure quota is enforcing workspace limits for %q", wsPath)
@@ -441,7 +444,7 @@ func TestClusterScopedQuota(t *testing.T) {
 			if err != nil {
 				return apierrors.IsForbidden(err), err.Error()
 			}
-			return true, ""
+			return false, "expected an error trying to create a workspace"
 		}, wait.ForeverTestTimeout, 100*time.Millisecond, "quota never rejected workspace creation in %q", wsPath)
 	}
 }

